go/vt/wrangler/traffic_switcher.go

```diff
@@ -59,6 +59,27 @@ const (
 	DirectionBackward
 )
 
+// TableRemovalType specifies the way the a table will be removed
+type TableRemovalType int
+
+// The following consts define if DropSource will drop or rename the table
+const (
+	DropTable = TableRemovalType(iota)
+	RenameTable
+)
+
+func (trt TableRemovalType) String() string {
+	types := [...]string{
+		"DROP TABLE",
+		"RENAME TABLE",
+	}
```

Comment:
Worth extracting the above code outside the method, since it always generates the same. Also, suggesting to make it a map, such that e.g. `types[DropTable]="DROP TABLE"`.

Reply:
@shlomi-noach from a logical perspective, it makes total sense to move the code outside a method that gets repeatedly called. However, I don't understand what you mean by "make it a map". I mean, I could figure out how, but I don't get why.

Reply:
@tomkrouper yeah, so I believe it would be a more idiomatic way; for example, right now the code runs `if trt < DropTable || trt > RenameTable {`, but, if we add a new, third option, then we'd need to change `trt > RenameTable` to `trt > SomethingNew` ; where in fact we don't care about ranges, but of existence, and that's where maps are idiomatic and their use is widespread.

So, if we have:

```go
mymap := map[string]string{DropTable: "DROP TABLE", RenameTable: :RENAME TABLE"}
```
we can then:
```go
if t, ok := mymap[trt] ; ok {
```
and that answers "is trt known or unknown to us". 

Anyway, that's the _why_ the way I understand it, but I may be just biased towards my own habits.

Reply:
Makes total sense. Thanks.